tests/test_cli.py

```diff
@@ -107,3 +108,69 @@ def test_create_user_config():
         "foo": None,
         "main": 3,
     }
+
+
+@pytest.fixture(name="authenticated_cli")
+def fixture_authenticated_cli():
+    def patched_get_auth_token():
+        return "token"
+
+    cli = AivenCLI()
+    cli._get_auth_token = patched_get_auth_token  # pylint: disable=protected-access
+    return cli
+
+
+def test_service_task_create_migration_check(authenticated_cli):
+    task_create_response = {
+        "message": "created",
+        "task": {
+            "create_time": "2021-08-13T08:00:45Z",
+            "result": "",
+            "success": None,
+            "task_id": "79803598-d09a-44bf-ae2b-34aad942f4e8",
+            "task_type": "mysql_migration_check"
+        }
+    }
+
+    with mock.patch("aiven.client.client.AivenClient") as cli_client_mock:
```

Comment:
Instead of using `mock.patch` and import paths, it would be less brittle to edit that part:
https://github.com/aiven/aiven-client/blob/1760ebfc970c00aa634db151d779165c5469f35e/aiven/client/cli.py#L106-L109
to add a `client_class = AivenClient` parameter to the init, and use it later to create the client.
this way a custom client can be injected when creating the `AivenCLI` instance in the `authenticated_cli` fixture.

Reply:
So basically the fixture would set `cli.client_class = Mock()` and then change the tests:

```diff
- cli_client_mock.return_value.get_service_task.return_value = task_get_response
+authenticated_cle.client_class.return_value.get_service_task.return_value = task_get_response
```
?

Looks a bit ugly, but returning a tuple `(authenticated_cli, client_mock)` looks just as ugly :-(

Reply:
Frankly speaking, I don't see why mocking of client is not appropriate here. I doubt adding `AivenClient` as a parameter is justifiable here, also how will it make tests less brittle? Can you give an example please? @kmichel-aiven

Reply:
Example here : https://github.com/aiven/aiven-client/commit/7eb65a07d28111f305ac61d90bf4a3e9b3db91fb

All variants I've tried have some degrees of ugliness, most of it caused by `argx.CommandLineTool.run` doing a lot of stuff all at once, I still think it's a first step toward making AivenCLI easier to test, and tested.

Reply:
Alright, I wouldn't say it look beautiful now, I'd prefer patching instead, but at this point I really don't mind. Will change as you suggest.